dart/neural/ConstrainedGroupGradientMatrices.cpp

```diff
@@ -1085,6 +1090,26 @@ Eigen::MatrixXs ConstrainedGroupGradientMatrices::getInvMassMatrix(
   return invMassMatrix;
 }
 
+Eigen::MatrixXs ConstrainedGroupGradientMatrices::getDampingDiagonal(
```

Comment:
As a trick to increase efficiency, you can have this return a `Eigen::VectorXs`, which is just the coefficients along the diagonal. Then wherever you use it, you can do `diagonalDampingVector.asDiagonal()`, which will make it behave like a matrix.

Reply:
Sure